app.js

```diff
@@ -3,24 +3,32 @@ const logger = require("morgan");
 const cors = require("cors");
 
 const contactsRouter = require("./routes/api/contacts");
-const userRouter = require("./controllers/testJWT"); // test
+const signup = require("./controllers/signup");
+const login = require("./controllers/login");
+const current = require("./controllers/current");
+const logout = require("./controllers/logout");
+const { middleware } = require("./controllers/middleware");
 
 const app = express();
-
 const formatsLogger = app.get("env") === "development" ? "dev" : "short";
 
 app.use(logger(formatsLogger));
 app.use(cors());
-app.use(express.json());
 
+app.use(express.json());
 app.use("/api/contacts", contactsRouter);
-app.use("/controllers/testJWT.js", userRouter); // test
 
-app.use((req, res) => {
+app.use(express.urlencoded());
+app.use("/api/users", signup);
+app.use("/api/users", login);
+app.use("/api/users", middleware, current);
+app.use("/api/users", middleware, logout);
```

Comment:
To powinno wyglądać trochę naczej. 
```suggestion
app.use("/api/users", usersRoutes);
```
Wewnątrze tego routera mieć wszystke ścieżki dotyczące uzytkownika. 

czyli tam będzie
`router.post('/signup', signup);`

Reply:
W ten sposób w app.js masz tylko zdefiniowane ścieżki ogólne, szczeóły zaś są w plikach gdzie są ścieżki.